### ISSUE TITLE
Problem with contract calls after version update

### ISSUE THREAD

**Issue by gperezalba** [graphprotocol/graph-node#2058]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
When some event triggers a contract call it spend very long time (2 to 5 minutes). The event doesn't appears in the logs of its respective block.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
I deployed 2 graph nodes (to load-balance them) in July. Some days ago I had to restart one of the servers containing one node. As I restarted the server I had to restart the docker-compose container containing the graph node. After that when some event triggers a contract call it spends a long time until the info in the subgraph is updated (2 to five minutes). Maybe the problem is with versions...my subgraphs were written with "@graphprotocol/graph-cli": "0.17.1" and "@graphprotocol/graph-ts": "0.17.0". As I restarted the graph node it updated to v.0.19.2 so maybe this is producing some unexpected behaviour.

I tried launching everything in a new server and same behaviour is observed.

EDIT: it seems the reason is the docker image of graph-node. I searched for the sha256 digest of the graph-node I had working correctly, then I searched for the tag of this sha256 digest and modified the docker-compose.yml file of the graph-node working bad and now it is updating info of contract calls on time (few seconds). Although it's working fine I would like to know the reason so I don't depend on a certain image.

Any help is appreciated. Thanks in advance.

**Comment by leoyvens** [comment]
@gperezalba Contract calls were switched to be made by block hash, utilizing EIP-1898. It may be that EIP-1898 calls are slower to process, if you're using the services of an Ethereum node provider, please check with them if there is a reason why EIP-1898 calls by hash would be slower than calls by number.

**Comment by gperezalba** [comment]
@leoyvens I'm almost sure this is the reason...I'm working with my own node because I'm running graph for a custom chain. I use parity and the version I'm running seems doesn't accept blockHash as a param. When I try any "eth_get..." RPC method it shows an error with message: `"Invalid params: invalid type: map, expected a block number or 'latest', 'earliest' or 'pending'."`. 

So taking into account that right now I cant update parity version...what can I do in Graph? I imagine that when graph-node cant call using blockHash it changes to use blockNumber/"latest" again...isn't it? Is there any way I can force the usage of blockNumber or "latest"?

Thanks :)

**Comment by leoyvens** [comment]
@gperezalba you can set the `GRAPH_ETH_CALL_BY_NUMBER` environment variable. This may lead to inconsistencies in the data, and it might not be supported forever. But it's there to go around existing limitations in the EIP-1898 implementation in Ethereum nodes.

**Comment by gperezalba** [comment]
@leoyvens I've tried using v.0.19.2 with `GRAPH_ETH_CALL_BY_NUMBER` setted to null or to 'latest' and it isn't working. I've checked the releases and I've found you included the calls by blockHash in v.0.19.0 so tried using v0.18.0 and doesn't work...I only can use graphprotocol/graph-node:8c54d02 dockerHub version to make it work (note it was relased 2 months after v0.18.0). 

The best would be to make it work with last version, but there must be something else appart from set `GRAPH_ETH_CALL_BY_NUMBER` (or maybe I setted it to a wrong value).
